### DISCUSSION TITLE
How to handle properties with `[ObservableProperty]` and the base override with `new`

### DISCUSSION THREAD

**Discussion by AndreasReitberger** [CommunityToolkit/dotnet#570] [Q&A] [answered]
I'm using a `ViewModelBase` which holds some propertie using `[ObservableProperty]`, like `IsPrinterOnline`.
In my `AppViewModel`, inherited from the `ViewModelBase` I need to override some base properties like shown below,

```cs
        [ObservableProperty]
        bool simulatePrint = true;

        bool _isPrinterOnline = false;
        public new bool IsPrinterOnline
        {
            get => _isPrinterOnline;
            set
            {
                if (_isPrinterOnline == value) return;
                _isPrinterOnline = value;
                OpenCurrentPrintPageCommand.ChangeCanExecute();
                OnPropertyChanged();
            }
        }
```

What's the best approach to use `[ObservableProperty]` on such `new` properties?

**Comment by Sergio0694** [comment, marked as answer]
Your example is not overriding a property, it's hiding one (shadowing). Is the question about how to get the `new` to be generated in the observable property? Otherwise I'm not entirely sure what the request is here 🤔

**EDIT:** this is not currently supported and it will be with https://github.com/CommunityToolkit/dotnet/issues/555.

**Reply by AndreasReitberger** [reply]
Yes, the question is how to get the `new` to the generated property, so that the shadowing is working (didn't know how to call, thanks for that!)

**Reply by Sergio0694** [reply]
This is not currently supported, it will be once https://github.com/CommunityToolkit/dotnet/issues/555 is available 🙂
